### PR TITLE
fix: don't report issue related to invalid position

### LINKED ISSUE #21: Issue found in/using golangci-lint, about warnings not processing properly
Please see this issue:

https://github.com/golangci/golangci-lint/issues/4550

We have concluded it's an issue in `contextcheck` but I do not know specifically what is going on. There is example code and changes made that can reproduce the issue:

https://github.com/golangci/golangci-lint/issues/4550#issuecomment-2009738678

### PR DESCRIPTION
**Pull request by ldez**
Fixes #21

### CODE REVIEW

File: contextcheck.go [hunk 1]
```diff
@@ -551,6 +559,10 @@ func (r *runner) checkFuncWithCtx(f *ssa.Function, tp entryType) {
 
 	for _, b := range f.Blocks {
 		for _, instr := range b.Instrs {
+			if !instr.Pos().IsValid() {
```
Comment (alexandear) [inline contextcheck.go]: Could we extend tests?

Perhaps the minimum reproducible example is the following:

```go
/* ----------------- issue 21 ----------------- */

func f16(ctx context.Context, k string) func() {
	return func() {
		f16(context.Background(), k)
	}
}
```

We can append it to the file https://github.com/kkHAIKE/contextcheck/blob/master/testdata/src/a/a.go

**Comment by kkHAIKE** [comment]
I really want to understand what kind of problem this is because in content check, each issue is quite important; otherwise, they cannot be connected.

Also, I hope this fix preferably doesn't alter the condition logic; just skipping Reportf will suffice.

```golang
func (r *runner) Reportf(pos token.Pos, format string, args ...interface{}) {
    if pos..IsValid() {
        r.pass.Reportf(pos, format, args...)
    }
}
```

**Comment by ldez** [comment]
I improved the PR to try to find the closest valid position.
I think that with this PR, 100% of reports will have a position (at least the closest possible to the problem).

File: testdata/src/a/a.go [hunk 2]
```diff
@@ -155,3 +155,19 @@ func f13[T int | int8 | int16 | int32 | int64 | uint | uint8 | uint16 | uint32 |
...
+	}
+}
+
+func f17(ctx context.Context, k string) func() func() {
+	return func() func() { // want "Function `f17\\$1->f17\\$1\\$1` should pass the context parameter"
+		return func() {
+			f16(context.Background(), k)
+		}
+	}
+}
```
Comment (alexandear) [inline testdata/src/a/a.go:173]: Maybe we should add a few cases covering HTTP handlers? Like

```go
func f18(w http.ResponseWriter, r *http.Request) func(w http.ResponseWriter, r *http.Request) {
	return func(w http.ResponseWriter, r *http.Request) {
		f8(context.Background(), w, r) // want "Non-inherited new context, use function like `context.WithXXX` or `r.Context` instead"
	}
}
```

**Review by alexandear — APPROVED** [review]

File: testdata/src/a/a.go (hunk 2 above)
Comment (ldez) [reply testdata/src/a/a.go:173]: This case was already working.

**Comment by kkHAIKE** [comment]
LGTM

**Pull request merged**